src/modules/commander/state_machine_helper.cpp

```diff
@@ -539,27 +536,27 @@ bool set_nav_state(vehicle_status_s &status, actuator_armed_s &armed, commander_
 			enable_failsafe(status, old_failsafe, mavlink_log_pub, reason_no_datalink);
 			set_link_loss_nav_state(status, armed, status_flags, internal_state, data_link_loss_act, 0);
 
-		} else if (status.rc_signal_lost && rc_loss_act_configured && status_flags.rc_signal_found_once
-			   && is_armed && !landed) {
-			// RC link lost, rc loss reaction configured, RC was used before -> RC loss reaction after delay
+		} else if (status.rc_signal_lost && !(param_com_rcl_except & RCLossExceptionBits::RCL_EXCEPT_MISSION)
+			   && status_flags.rc_signal_found_once && is_armed && !landed) {
+			// RC link lost, rc loss not disabled in mission, RC was used before -> RC loss reaction after delay
 			// Safety pilot expects to be able to take over by RC in case anything unexpected happens
 			enable_failsafe(status, old_failsafe, mavlink_log_pub, reason_no_rc);
 			set_link_loss_nav_state(status, armed, status_flags, internal_state, rc_loss_act, param_com_rcl_act_t);
 
-		} else if (status.rc_signal_lost && rc_loss_act_configured
+		} else if (status.rc_signal_lost && !(param_com_rcl_except & RCLossExceptionBits::RCL_EXCEPT_MISSION)
 			   && status.data_link_lost && !data_link_loss_act_configured
 			   && is_armed && !landed) {
 			// All links lost, no data link loss reaction configured -> immediately do RC loss reaction
 			// Lost all communication, by default it's considered unsafe to continue the mission
-			// Note this case is reached after the previous one when flying mission completely without RC
+			// This is only reached when flying mission completely without RC (it was not present since boot)
 			enable_failsafe(status, old_failsafe, mavlink_log_pub, reason_no_datalink);
 			set_link_loss_nav_state(status, armed, status_flags, internal_state, rc_loss_act, 0);
 
-		} else if (status.rc_signal_lost && !rc_loss_act_configured
+		} else if (status.rc_signal_lost && (param_com_rcl_except & RCLossExceptionBits::RCL_EXCEPT_MISSION)
 			   && status.data_link_lost && !data_link_loss_act_configured
 			   && is_armed && !landed
 			   && mission_finished) {
-			// All links lost, all link loss reactions disabled -> return after mission
+			// All links lost, all link loss reactions disabled -> return after mission finished
 			// Pilot disabled all reactions, finish mission but then return to avoid lost vehicle
 			enable_failsafe(status, old_failsafe, mavlink_log_pub, reason_no_datalink);
```

Comment:
Should there be a `reason_no_rc_and_no_datalink` here?

Reply:
There should. I would call it `reason_all_links_lost` but maybe your suggestion is more consistent.

Reply:
Ah `reason_all_links_lost` works.

Reply:
![image](https://user-images.githubusercontent.com/4668506/123104790-8e435800-d437-11eb-9ad7-4465d5924b8e.png)

Reply:
Adressed in https://github.com/PX4/PX4-Autopilot/pull/17785/commits/72543a94b56dd9a46cc2f3e871ef80bc250c5bf8